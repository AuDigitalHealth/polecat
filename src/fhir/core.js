import pick from 'lodash.pick'

import { OpOutcomeError } from '../OpOutcomeError.js'

export const opOutcomeFromJsonResponse = parsed => {
  if (parsed.resourceType !== 'OperationOutcome') return null
  if (parsed.issue.length === 0) return null
  return new OpOutcomeError(
    pick(
      // We only ever look at the first issue described within an
      // OperationOutcome resource.
      parsed.issue[0],
      'severity',
      'code',
      'details',
      'diagnostics',
      'location',
      'expression',
    ),
  )
}
