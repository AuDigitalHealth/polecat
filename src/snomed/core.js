import { parseUri } from '../util.js'

export const snomedUri = 'http://snomed.info/sct'

export const auEditionId = '32506021000036107'
export const internationalEditionId = '900000000000207008'

export const humaniseUri = uri => {
  return uri === snomedUri ? 'SNOMED CT' : uri
}

export const humaniseVersion = (uri, version) => {
  if (uri !== snomedUri) return version
  const parsed = parseUri(version),
    versionUri = decodeURIComponent(parsed.queryKey.version),
    path = versionUri.split(snomedUri + '/')[1],
    [edition, release] = path.split('/version/'),
    editionDisplay = {
      [auEditionId]: 'Australian',
      [internationalEditionId]: 'International',
    }[edition]
  if (!edition) return versionUri
  return `${editionDisplay} Edition (${release})`
}
