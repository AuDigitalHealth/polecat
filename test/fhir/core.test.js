import { opOutcomeFromJsonResponse } from '../../src/fhir/core.js'

import opOutcome from '../fixtures/operationOutcome.json'

describe('opOutcomeFromJsonResponse', () => {
  it('should return a correct OpOutcomeError', () =>
    expect(opOutcomeFromJsonResponse(opOutcome)).toMatchSnapshot())
})
