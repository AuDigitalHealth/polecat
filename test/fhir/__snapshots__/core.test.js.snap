// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opOutcomeFromJsonResponse should return a correct OpOutcomeError 1`] = `[OpOutcomeError: Invalid request: The FHIR endpoint on this server does not know how to handle GET operation[Medication] with parameters [[_text, is-brafnd]]]`;
